eng/Generate.ps1

```diff
@@ -60,6 +63,21 @@ function Add-CadlRanch-Typespec([string]$testName, [string]$projectPrefix, [stri
     Add-Typespec "$projectPrefix$testName" $projectDirectory $cadlMain
 }
 
+function Get-TypeSpec-Entry([System.IO.DirectoryInfo]$directory) {
+    $clientPath = Join-Path $directory "client.tsp"
+    if (Test-Path $clientPath) {
+        return $clientPath
+    }
+
+    $mainPath = Join-Path $directory "main.tsp"
+    if (Test-Path $mainPath) {
+        return $mainPath
+    }
+
+    $projectNamePath = Join-Path $directory "$($directory.Name).tsp"
```

Comment:
please check the existence of this file also, and throw error if it does not exist.